### PR TITLE
Add test to check that all settings frames are acked

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
ping

**Comment by dgquintas** [comment]
Hmm I don't quite speak Go. Do you still trust me reviewing?

**Comment by carl-mastrangelo** [comment]
Yes, if you have questions about the syntax I would be happy to answer them.  Go's syntax was specifically designed to appeal to existing C / Java programmers.

**Comment by dgquintas** [comment]
Yes, I also went through the tour back in 2009, just a couple yea... WHAT, IT'S ALMOST 2016?!

ok gimme a sec

File: tools/http2_interop/s6.5.go [hunk 1]
```diff
@@ -30,3 +31,60 @@ func testSmallMaxFrameSize(ctx *HTTP2InteropCtx) error {
...
+		f, err := parseFrame(conn)
+		if err != nil {
+			return err
+		}
+
+		// Other frames come down the wire too, including window update.  Just ignore those.
+		if f, ok := f.(*SettingsFrame); ok {
+			settingsFramesReceived += 1
+			if settingsFramesReceived == 1 {
+				if f.Header.Flags&SETTINGS_FLAG_ACK > 0 {
```
Comment (dgquintas) [inline tools/http2_interop/s6.5.go]: no space around the & operator?

File: tools/http2_interop/s6.5.go [hunk 2]
```diff
@@ -30,3 +31,60 @@ func testSmallMaxFrameSize(ctx *HTTP2InteropCtx) error {
...
+					return fmt.Errorf("settings frame should not have used ack: %v")
+				}
+				continue
+			}
+
+			if f.Header.Flags&SETTINGS_FLAG_ACK == 0 {
```
Comment (dgquintas) [inline tools/http2_interop/s6.5.go]: ditto

**Comment by dgquintas** [comment]
LGTM other than the minor stylistic comment/question.

File: tools/http2_interop/s6.5.go (hunk 1 above)
Comment (carl-mastrangelo) [inline tools/http2_interop/s6.5.go]: Yes, Go uses a formatter that automatically formats code this way.  The spacing is there to make it obvious that the & operator is higher precedence than >.  (I think)

**Pull request merged**